packages/smaz-compress/src/index.ts

```diff
@@ -13,11 +13,16 @@ export class SmazCompress {
     this.verbatim = new Uint8Array(255);
   }
 
-  public getCompressedSize(str: string): number {
+  public getCompressedSize(str: string | Uint8Array): number {
     if (str.length === 0) {
       return 0;
     }
 
+    const retrieve =
+      typeof str === 'string'
+        ? str.charCodeAt.bind(str)
+        : (str.at.bind(str) as (i: number) => number);
```

Comment:
Nit. Maybe for typing it could be more elegant to use lambdas? 
```suggestion
        ? (idx: number): number => str.charCodeAt(str)
        : (idx: number): number => str[idx];
```